driver/configuration/provider_viper.go

```diff
@@ -86,35 +86,9 @@ const (
 	ViperKeyAuthenticatorUnauthorizedIsEnabled = "authenticators.unauthorized.enabled"
 )
 
-func BindEnvs() {
-	if err := viper.BindEnv(
-		ViperKeyProxyReadTimeout,
```

Comment:
Removing these will have a detrimental effect. Viper does not automatically check environment variables. You have to bind them explicitly.

Reply:
> Viper does not automatically check environment variables. You have to bind them explicitly.

Why not? It indeed loads envs automatically: https://github.com/spf13/viper/blob/master/viper.go#L1039

In hydra, everything works fine without any explicit bindings.

Reply:
Your analysis is unfortunately incorrect. In Hydra, we do not (yet) use JSON Schema to validate the config. However, validating against the Config JSON Schema requires all keys to be loaded. That's just the way Viper works (it is quirky).

Reply:
Sorry, I do not understand how we suppose to explicitly bind all keys, since most of them were recently removed for example in https://github.com/ory/oathkeeper/pull/258 . How do envs like `AUTHENTICATORS_OAUTH2_INTROSPECTION_CONFIG_PRE_AUTHORIZATION_CLIENT_ID` suppose to work?

(also related to the comment below about tests)

Reply:
While the keys have been removed as constants (because we don't explicitly need them any more - except for `BindEnvs()`). However, I think it would be better to add a patch to our fork to automatically bind all envs when doing `AutomaticEnv()` instead of doing both `AutomaticEnv()` as well as `BindEnv()`. This is a known (and unsolved issue) in viper that has lead to some frustration (me included): https://github.com/spf13/viper/issues/761 (and all linked issues).

Then, we can throw away `BindEnvs()` which - to be honest - sucks anyways.

Reply:
This however would need some thinking as `_` can both be `.` as well as `_`. So we'd end up with a lot of keys when "autom-mounting" all environment keys. For example env var `FOO_BAR` could both be `foo_bar` as well as `foo.bar`. That would in turn cause issues when we set `additionalProperties: false` in the JSON Config Schema - which is not great because with that property set people would immediately find any typos they're doing. Maybe by defining some "categories" that always replace `_` with `.` would be the way to go. Not sure yet, needs some thinking.

Reply:
Actually, I just found my original issue which touched this topic in 2016 :D https://github.com/spf13/viper/issues/212

Reply:
Oh, I see, things getting complicated, combining viper behaviour and the introduction of json schemas.

If you come up with the idea how should we solve it, let me know.

I've pushed changes that make env vars work for defined keys (e.g. `SERVE_PROXY_PORT`), but envs like `AUTHENTICATORS_OAUTH2_INTROSPECTION_CONFIG_PRE_AUTHORIZATION_CLIENT_ID` still do not work. If you feel like we don't need this intermediary improvement, feel free to close this MR.